junit-platform-launcher/src/main/java9/org/junit/platform/launcher/listeners/FlightRecordingListener.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.listeners;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Event;
+import jdk.jfr.Label;
+
+import org.apiguardian.api.API;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestIdentifier;
+import org.junit.platform.launcher.TestPlan;
+
+/**
+ * A {@link TestExecutionListener} that generates Java Flight Recorder
+ * events.
+ *
+ * @see <a href="https://openjdk.java.net/jeps/328">JEP 328: Flight Recorder</a>
+ * @since 1.7
+ */
+@API(status = EXPERIMENTAL, since = "1.7")
+public class FlightRecordingListener implements TestExecutionListener {
+
+	private final AtomicReference<TestPlanExecutionEvent> testPlanExecutionEvent;
+	private final Map<String, TestExecutionEvent> testExecutionEventMap;
+
+	public FlightRecordingListener() {
+		this.testPlanExecutionEvent = new AtomicReference<>();
+		this.testExecutionEventMap = new ConcurrentHashMap<>();
```

Comment:
Why not assign field directly, i.e. testExecutionEventMap = new ConcurrentHashMap<>()?

Reply:
I like field declarations to be separated from their assignment.

The constructor is a good place to do that work, which not much here, yet. But you never know.

Reply:
Why assign a field directly? Performance?

Reply:
Less code.